amlb/benchmark.py

```diff
@@ -116,7 +116,9 @@ def setup(self, mode: SetupMode):
 
         log.info("Setting up framework {}.".format(self.framework_name))
 
-        self._write_setup_env(self.framework_module.__path__[0], **dict(self.framework_def.setup_env))
+        framework_def_plus_user_dir_dict = self.framework_def.setup_env
+        framework_def_plus_user_dir_dict['USER_DIR'] = rconfig().user_dir
+        self._write_setup_env(self.framework_module.__path__[0], **dict(framework_def_plus_user_dir_dict))
```

Comment:
if you just need to add some constants, please add them to `_write_setup_env` definition: it already sets some hardcoded defaults